tests/astronomer/flask_appbuilder/test_security.py

```diff
@@ -0,0 +1,152 @@
+from astronomer.flask_appbuilder.security import AstroSecurityManagerMixin
+
+
+AUDIENCE = 'airflow.example.com'
+
+
+@pytest.fixture(scope='module')
+def app():
+    from flask import Flask
+    app = Flask(__name__)
+    app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+    app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///:memory:'
+    app.config['SECRET_KEY'] = 'thisismyscretkey'
+    app.config['TESTING'] = True
+
+    @app.route("/")
+    def home():
+        return "Hello"
+    return app
+
+
+@pytest.fixture(scope='module')
+def appbuilder(app, sm_class):
+    from flask_appbuilder import AppBuilder, SQLA
+
+    db = SQLA(app)
+
+    appbuilder = AppBuilder(app, db.session, security_manager_class=sm_class)
+
+    for r in ('Admin', 'Op', 'User', 'Viewer'):
+        appbuilder.sm.add_role(r)
+
+    """
+    from sqlalchemy import event
+    @event.listens_for(db.engine, 'before_cursor_execute')
+    def before_cursor_execute(conn, cursor, statement, parameters,
+                              context, executemany):
+        print("Received statement: %s", statement)
+        print("Received parameters: %r", parameters)
+    """
+
+    return appbuilder
+
+
+@pytest.fixture
+def user(appbuilder):
+    username = 'airflower'
+    role = appbuilder.sm.find_role('Admin')
+    if not appbuilder.sm.add_user(username, 'Lucy', 'Airflower', 'airflow@domain.com', role):
+        raise RuntimeError("Error creating test user")
+    user = appbuilder.sm.find_user(username=username)
+    yield user
+    appbuilder.session.delete(user)
+
+
+@pytest.fixture(scope='module')
+def sm_class(jwt_signing_key):
+    class SM(AstroSecurityManagerMixin, flask_appbuilder.security.sqla.manager.SecurityManager):
+        def count_users(self):
+            # Silence the log message about no users
+            return 1
+
+    # AppBuilder needs a function that accepts a single argument (the AppBuilder instance)
+    return lambda appbuilder: SM(appbuilder, jwt_signing_key, allowed_audience=AUDIENCE)
+
+
+@pytest.fixture(scope='session')
+def jwt_signing_key():
+    return jwk.JWK(generate='oct', size=256)
+
+
+@pytest.fixture
+def invalid_jwt():
+    ephemeral_key = jwk.JWK(generate='oct', size=64)
+    token = jwt.JWT(header={"alg": "HS256"}, claims={"info": "I'm a signed token"})
+    token.make_signed_token(ephemeral_key)
+    return token.serialize()
+
+
+@pytest.fixture
+def signed_jwt(jwt_signing_key):
+    def jwt_factory(claims):
+        token = jwt.JWT(header={"alg": "HS256"}, claims=claims)
+        token.make_signed_token(jwt_signing_key)
+        return token.serialize()
+    return jwt_factory
+
+
+@pytest.fixture(params=[['Op']])
+def valid_claims(request):
+    return {
+        'email': 'airflow@domain.com',
+        'roles': request.param,
+        'sub': 'airflow@domain.com',
+        'full_name': 'Lucy Airflower',
+        'aud': AUDIENCE,
+    }
+
+
+@pytest.mark.usefixtures('client_class')
+class TestAstroSecurityManagerMixin:
```

Comment:
better to split tests in smaller functions, e.g.
https://github.com/aio-libs/aiohttp/blob/master/tests/test_client_connection.py

Reply:
I usually go with "one test class per python class we're testing" - that way  I don't have to prefix every test with the class name too.

(There will also be a `class TestAirflowAstroSecurityManager` coming)

Reply:
ok with me.